### ISSUE TITLE
Problems with the httpyac.config.js file

### ISSUE THREAD

**Issue by alekdavisintel** [AnWeber/vscode-httpyac#290]
I have a working project (very simple, just a couple of tests). I added an `httpyac.config.js` file to the project folder and copied the content as-is from the sample at https://httpyac.github.io/config/#httpyac-config-js. The project file structure looks like this:

![image](https://github.com/AnWeber/vscode-httpyac/assets/98194335/17875c63-2a1b-44b6-8488-2b60aca4cc30)

Once I added the config file, I immediately got an error: _ReferenceError: models is not defined at Object.&lt;anonymous&gt;_. I assumed it corresponded to the `level: models.LogLevel.warn` setting in the file, so I removed it. The error disappeared, but now the the requests were hanging. I suspected it to be a proxy issue, so I added the `proxy` setting to the config file, and it resolved the hanging request problem (I assumed that if the setting is not present in the file, the default would be used, but it looks like I was wrong). Now, when I run tests, I get a file not found warning (at the bottom right corner).

![image](https://github.com/AnWeber/vscode-httpyac/assets/98194335/33d7a759-d4ea-4aea-9625-4ca739ee992c)

The test complete without issues, so I'm not sure why the warning pops up, although the warning makes sense because my import statement looks like this :

```JavaScript
# @import ../../../common.http
```
Logically, the import statement should use two levels up, not three, i.e. `../../common.http`, but if I change it to use two levels up, none of my variables defined in the `common.http` file get resolved and the tests fail with the _baseUrl is not defined_ error:

![image](https://github.com/AnWeber/vscode-httpyac/assets/98194335/c52477d4-cd68-45d0-83f2-0c1deafde30d)

Any ideas what is going on here?

P.S. I removed the `httpyac.config.js` file from the project, bu the _file not found_ warning still shows (the tests still run fine), so I'm not sure if it is config file related (before I started messing with the config file, I did not get the warning, but maybe I did something, while trying different things).

**Comment by AnWeber** [comment]
I must confess that I can't manage the problems you describe. The `ReferenceError` is clear because I accidentally wrote Typescript code in Javascript (example is corrected). But then I can't manage the subsequent problems. Requests work for me even without setting the proxy. The inline error with `baseUrl not defined` does not come from my extension. I did not implement some kind of inline error handling, only popups
The `File not found` warning comes from my extension, but I have decided not to prevent the execution on such an error. If the variables are already present, the requests would still go through. The handling of variables in the VSCode extension is relatively lax to allow easy use. VSCode simply uses all variables that have already been used once before and also caches them between executions. Setting [`httpyac.regionScopedVariables`](https://github.com/AnWeber/vscode-httpyac/blob/main/package.json#L551) would change behaviour or reset Environment between runs.

[test.zip](https://github.com/AnWeber/vscode-httpyac/files/15364136/test.zip)

**Comment by alekdavisintel** [comment]
Thank you for the sample. I installed it as-is (although, had to change the `host2` property in the environment file to `host` because with `host2`, I got an error complaining about the endpoint URL). It ran fine when I was not on the corporate VPN, but as soon as I got on the VPN, it failed because it could not reach the endpoint. I turned VPN off and it worked fine again. Then I turned VPN on again, closed Visual Studio Code, renamed the `.httpyac.config.js` file to `.httpyac.config.js.BAK`, relaunched  Visual Studio Code, and the request worked (with VPN turned on). 

To summarize: when using a config file, if a proxy setting was not explicitly defined in the file, requests over VPN did not work, but without the config file they did. I do not remember if I configured global proxy setting for httpYAC somewhere when I installed it. I do not think so, but maybe I forgot. And what it looks like to me is that when a config file is used, proxy settings are treated differently. It's not a big deal, but a bit weird. 

I should mention that our proxy settings are configured in a setup script that is pretty complicated (I do not understand details, but it uses the client and the destination URL to determine if  a proxy is needed and, if so, which one, since clients in India would use a different proxy servers from client in Europe or North America). It looks something like this (with corporate specific info removed):

```JavaScript
function FindProxyForURL(url,host) {
var me=myIpAddress();
var resolved_ip = dnsResolve(host);
if (host == "127.0.0.1") {return "DIRECT";}
if (host == "localhost") {return "DIRECT";}
if (isPlainHostName(host)) {return "DIRECT";}
if (url.substring(0,37) == "http://abc.def.com") {return "DIRECT";}
if (!resolved_ip) {return "PROXY xxx.xxx.com:XXX";}
// 100 more if statements
// ...
return "PROXY yyy.xxx.com:XXX";
}
```
I figure out the missing file problem (thanks for the sample again because it helped my get on the right path). I had too many up levels in my `import` statement, so once I changed `@import ../../../common.http` to `@import ../common.http` the warning disappeared. I am surpised it worked at all with `../../..`, so let's consider it a miracle. :-)

Thank you so much for all your help. I just love your extension. It is a pleasure to use (although, I am still hazy about some areas, but I will try to figure them out).  I originally wanted to find something that would understand our original Postman implementation, but after working with httpYAC for a couple of days, I like it's architecture a lot better, so I do not mind doing a conversion. Once I finish my POC, assuming I will not run into any show stoppers, I will try to push httpYAC for adoption across the IT and other organizations. 

Thanks again. I really appreciate your work and all the help.

**Comment by AnWeber** [comment]
The different behavior irritates me. It should not be due to the proxy PAC. I didn't have any problems with one in my previous work. I also couldn't see any difference in the configuration if the file is present or not. 
With setting `httpyac.logLevel=debug` you can see in the OutputChannel `httpyac.log` which proxy is used ([search for request](https://github.com/AnWeber/httpyac/blob/main/src/plugins/http/gotUtils.ts#L59-L62))
<img width="1008" alt="image" src="https://github.com/AnWeber/vscode-httpyac/assets/13434739/a11d3841-ca60-4a94-b6ba-3dde9275e6da">

VSCode may also intervene here, as it injects code when the proxy is used. For example, http2 will not work with the default setting of vscode for this reason. You could also test and change the settings there ([`http.proxySupport`](https://github.com/microsoft/vscode/blob/7c8252c59b8025c05bdeb10eb62d1b7e1c3fa9dd/src/vs/platform/request/common/request.ts#L164)). Or use httpyac CLI to test without vscode.
It would be a pity if you could not get the file to run. It would be interesting for your own plugins that you might need for your use case.

**Comment by alekdavisintel** [comment]
I just retried it to compre the logs and it worked fine. Not sure what I did differently. I forgot to select an environment a few times after restarting VSC but I think I did realize it at some point and re-ran (still can't get used to the fact that environment is not remembered between the sessions, and I know there is a way to define the default environment, but I think it's global, and between the projects, environment names may differ, so a global default environment doe not really make much sense. and I did not see an option to define project/folder-specific default environment, but maybe I missed it). Anyway, I will keep an eye on it and look at debug logs if I see it again. Thanks.
